?  How to Update from CVS.txt
? build-release
? build-wxw.sh
? mac_native_splines.patch
? contrib/samples/deprecated/proplist/.DS_Store
? contrib/samples/deprecated/resource/.DS_Store
? contrib/samples/deprecated/treelay/.DS_Store
? demos/bombs/.DS_Store
? demos/dbbrowse/.DS_Store
? demos/forty/.DS_Store
? demos/fractal/.DS_Store
? demos/life/.DS_Store
? demos/poem/.DS_Store
? include/.DS_Store
? include/wx/.DS_Store
? include/wx/setup.h
? include/wx/mac/.DS_Store
? include/wx/mac/carbon/.DS_Store
? samples/.DS_Store
? samples/access/.DS_Store
? samples/artprov/.DS_Store
? samples/calendar/.DS_Store
? samples/caret/.DS_Store
? samples/checklst/.DS_Store
? samples/config/.DS_Store
? samples/controls/.DS_Store
? samples/db/.DS_Store
? samples/dialogs/.DS_Store
? samples/dnd/.DS_Store
? samples/docview/.DS_Store
? samples/docview/Carbon_MachO(2.5.3).app
? samples/docview/Carbon_MachO.app
? samples/docview/Carbon_MachODebug(2.5.3).app
? samples/docview/Carbon_MachODebug.app
? samples/docview/docview.plc
? samples/docview/docviewM8.mcp
? samples/docvwmdi/.DS_Store
? samples/dragimag/.DS_Store
? samples/drawing/.DS_Store
? samples/dynamic/.DS_Store
? samples/erase/.DS_Store
? samples/event/.DS_Store
? samples/exec/.DS_Store
? samples/font/.DS_Store
? samples/grid/.DS_Store
? samples/help/.DS_Store
? samples/html/about/.DS_Store
? samples/html/help/.DS_Store
? samples/html/helpview/.DS_Store
? samples/html/printing/.DS_Store
? samples/html/test/.DS_Store
? samples/html/virtual/.DS_Store
? samples/html/widget/.DS_Store
? samples/html/zip/.DS_Store
? samples/image/.DS_Store
? samples/internat/.DS_Store
? samples/joytest/.DS_Store
? samples/keyboard/.DS_Store
? samples/layout/.DS_Store
? samples/listbox/.DS_Store
? samples/listctrl/.DS_Store
? samples/mdi/.DS_Store
? samples/mediaplayer/.DS_Store
? samples/menu/.DS_Store
? samples/minifram/.DS_Store
? samples/minimal/.DS_Store
? samples/minimal/Carbon_MachO.app
? samples/minimal/Carbon_MachODebug.app
? samples/minimal/Carbon_MachO_Shlb.app
? samples/minimal/Carbon_MachO_Shlb_Debug.app
? samples/minimal/minimalM8.mcp
? samples/notebook/.DS_Store
? samples/opengl/cube/.DS_Store
? samples/opengl/isosurf/.DS_Store
? samples/opengl/penguin/.DS_Store
? samples/ownerdrw/.DS_Store
? samples/png/.DS_Store
? samples/printing/.DS_Store
? samples/printing/printing_original.cpp
? samples/propsize/.DS_Store
? samples/regtest/.DS_Store
? samples/richedit/.DS_Store
? samples/rotate/.DS_Store
? samples/sashtest/.DS_Store
? samples/scroll/.DS_Store
? samples/scrollsub/.DS_Store
? samples/shaped/.DS_Store
? samples/splitter/.DS_Store
? samples/statbar/.DS_Store
? samples/taskbar/.DS_Store
? samples/text/.DS_Store
? samples/thread/.DS_Store
? samples/toolbar/.DS_Store
? samples/treectrl/.DS_Store
? samples/typetest/.DS_Store
? samples/validate/.DS_Store
? samples/widgets/.DS_Store
? samples/wizard/.DS_Store
? src/wxWindowsM8 Data
? src/wxWindowsM8.mcp
? src/wxWindowsM9 Data
? src/wxWindowsM9.mcp
? src/common/.DS_Store
? src/jpeg/jpegM8.mcp
? src/mac/.DS_Store
? src/mac/carbon/.DS_Store
? src/png/pngM8.mcp
? src/regex/.DS_Store
? src/regex/regexM8 Data
? src/regex/regexM8.mcp
? src/tiff/tiffM8.mcp
? src/wxWindows.xcodeproj/rjlang.mode1
? src/zlib/zlibM8.mcp
? utils/HelpGen/src/.DS_Store
? utils/helpview/src/.DS_Store
? utils/tex2rtf/src/.DS_Store
Index: include/wx/mac/carbon/dc.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/carbon/dc.h,v
retrieving revision 1.11
diff -b -u -2 -r1.11 dc.h
--- include/wx/mac/carbon/dc.h	2005/02/18 16:45:02	1.11
+++ include/wx/mac/carbon/dc.h	2005/09/12 18:46:20
@@ -55,4 +55,6 @@
     virtual void AddLineToPoint( wxCoord x1 , wxCoord y1 ) = 0 ;
     
+    virtual void AddQuadCurveToPoint( wxCoord cx1, wxCoord cy1, wxCoord x1, wxCoord y1 ) = 0 ;
+    
     virtual void AddRectangle( wxCoord x, wxCoord y, wxCoord w, wxCoord h ) = 0 ;
     
@@ -238,4 +240,7 @@
 
     virtual void DoDrawPoint(wxCoord x, wxCoord y);
+#if wxUSE_SPLINES
+    virtual void DoDrawSpline(wxList *points);
+#endif // wxUSE_SPLINES
     virtual void DoDrawLine(wxCoord x1, wxCoord y1, wxCoord x2, wxCoord y2);
 
Index: include/wx/mac/carbon/private.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/carbon/private.h,v
retrieving revision 1.39
diff -b -u -2 -r1.39 private.h
--- include/wx/mac/carbon/private.h	2005/04/24 07:53:56	1.39
+++ include/wx/mac/carbon/private.h	2005/09/12 18:46:21
@@ -600,4 +600,5 @@
     void MoveToPoint( wxCoord x1 , wxCoord y1 ) ;
     void AddLineToPoint( wxCoord x1 , wxCoord y1 ) ;
+    void AddQuadCurveToPoint( wxCoord cx1, wxCoord cy1, wxCoord x1, wxCoord y1 ) ;
     void AddRectangle( wxCoord x, wxCoord y, wxCoord w, wxCoord h ) ;
     void AddCircle( wxCoord x, wxCoord y , wxCoord r ) ;
Index: src/mac/carbon/dccg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/dccg.cpp,v
retrieving revision 1.25
diff -b -u -2 -r1.25 dccg.cpp
--- src/mac/carbon/dccg.cpp	2005/09/01 06:36:21	1.25
+++ src/mac/carbon/dccg.cpp	2005/09/12 18:46:34
@@ -158,4 +158,9 @@
 }
 
+void wxMacCGPath::AddQuadCurveToPoint( wxCoord cx1, wxCoord cy1, wxCoord x1, wxCoord y1 )
+{
+    CGPathAddQuadCurveToPoint( m_path , NULL , cx1 , cy1 , x1 , y1 );
+}
+
 void wxMacCGPath::AddRectangle( wxCoord x, wxCoord y, wxCoord w, wxCoord h )
 {
@@ -1421,4 +1426,63 @@
     delete path ;
 }
+
+#if wxUSE_SPLINES
+void wxDC::DoDrawSpline(wxList *points)
+{
+    wxCHECK_RET(Ok(), wxT("Invalid DC"));
+    
+    if ( m_logicalFunction != wxCOPY )
+        return ;
+
+    wxGraphicPath* path = m_graphicContext->CreatePath() ;
+
+    wxList::compatibility_iterator node = points->GetFirst();
+    if (node == wxList::compatibility_iterator())
+        // empty list
+        return;
+
+    wxPoint *p = (wxPoint *)node->GetData();
+
+    wxCoord x1 = p->x;
+    wxCoord y1 = p->y;
+
+    node = node->GetNext();
+    p = (wxPoint *)node->GetData();
+
+    wxCoord x2 = p->x;
+    wxCoord y2 = p->y;
+    wxCoord cx1 = ( x1 + x2 ) / 2;
+    wxCoord cy1 = ( y1 + y2 ) / 2;
+
+    path->MoveToPoint( XLOG2DEVMAC( x1 ) , XLOG2DEVMAC( y1 ) ) ;
+    path->AddLineToPoint( XLOG2DEVMAC( cx1 ) , XLOG2DEVMAC( cy1 ) ) ;
+
+#if !wxUSE_STL
+    while ((node = node->GetNext()) != NULL)
+#else
+    while ((node = node->GetNext()))
+#endif // !wxUSE_STL
+    {
+        p = (wxPoint *)node->GetData();
+        x1 = x2;
+        y1 = y2;
+        x2 = p->x;
+        y2 = p->y;
+        wxCoord cx4 = (x1 + x2) / 2;
+        wxCoord cy4 = (y1 + y2) / 2;
+
+        path->AddQuadCurveToPoint( XLOG2DEVMAC( x1 ) , XLOG2DEVMAC( y1 ) , 
+          XLOG2DEVMAC( cx4 ) , XLOG2DEVMAC( cy4 ) ) ;
+
+        cx1 = cx4;
+        cy1 = cy4;
+    }
+
+    path->AddLineToPoint( XLOG2DEVMAC( x2 ) , XLOG2DEVMAC( y2 ) ) ;
+
+    m_graphicContext->StrokePath( path ) ;
+    delete path ;
+}
+#endif
 
 void  wxDC::DoDrawPolygon(int n, wxPoint points[],

 	  	 
